### ISSUE TITLE
flexible loss function swapping

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#282]
the opensoundscape.torch.loss module and cnn module are set up to allow choice of loss functions, but it doesn't quite work yet to choose built-in loss functions from pytorch's torch.nn. This is probably because of one-hot vs labeled encoding or because of a type error (needing to perform target = target.float() as in BCEWithLogitsLoss_hot). 

This should be a pretty easy fix and will allow us to easily swap out loss functions.

**Comment by sammlapp** [comment]
In general, there isn't much of a reason to change the loss function. Single target problems including 2-class problems should use CrossEntropyLoss (LogSoftmax + NLL loss) while multi-target problems should use BCEWithLogitsLoss, or a variant.
